src/extension/background-script/actions/nostr/enable.ts

```diff
@@ -0,0 +1,84 @@
+import utils from "~/common/lib/utils";
+import { getHostFromSender } from "~/common/utils/helpers";
+import db from "~/extension/background-script/db";
+import type { MessageAllowanceEnable, Sender } from "~/types";
+
+import state from "../../state";
+import { ExtensionIcon, setIcon } from "../setup/setIcon";
+
+const enable = async (message: MessageAllowanceEnable, sender: Sender) => {
+  const host = getHostFromSender(sender);
+  if (!host) return;
+
+  const isUnlocked = await state.getState().isUnlocked();
+  const account = await state.getState().getAccount();
+  const allowance = await db.allowances
+
+    .where("host")
+    .equalsIgnoreCase(host)
+    .first();
+
+  if (
+    isUnlocked &&
+    allowance &&
+    allowance.enabled &&
+    account?.nostrPrivateKey
```

Comment:
why is the `nostrPrivateKey` check needed?

Reply:
That's to force onboarding in the enable prompt, right?

Reply:
this is needed because if allowance is set we directly return from the prompt itself. we don't go to screen

Reply:
we need to open enable screen and render screen conditionally